test/built-ins/Temporal/PlainDateTime/prototype/with/options-empty.js

```diff
@@ -0,0 +1,23 @@
+// Copyright (C) 2022 Igalia, S.L. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-temporal.plaindatetime.prototype.with
+description: Verify that undefined options are handled correctly.
+features: [Temporal]
+includes: [temporalHelpers.js]
+---*/
+
+const datetime = new Temporal.PlainDateTime(1976, 11, 18, 15, 23, 30, 123, 456, 789);
+
+TemporalHelpers.assertPlainDateTime(
+  datetime.with({ day: 5 }, {}),
```

Comment:
More interesting with:
```suggestion
  datetime.with({ day: 40 }, {}),
```

Same below.